### PR TITLE
Allow using `put` as a source

### PR DESCRIPTION
**Pull request by dominiklohmann**
This changes `put` so that it can be used to inject a static event into a pipeline.

### CODE REVIEW

File: changelog/next/features/3669--put-as-asource.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The `put` operator is now a source operator as well, which allows for
+introducing one event into a pipeline with known values.
```
Comment (dominiklohmann) [inline changelog/next/features/3669--put-as-asource.md:2]: @jachris and I discussed on Thursday that this is quite the breaking change to make because of implicit sources, and that we'd rather introduce a new operator for this. I agree with this fundamentally. @mavam, do you have any naming ideas?

File: libtenzir/include/tenzir/concept/parseable/tenzir/data.hpp [hunk 2]
```diff
@@ -71,11 +71,11 @@ struct data_parser : parser_base<data_parser> {
     // with named fields. Allowing a mixture of both would mean we'd
     // have to deal with ambiguous inputs.
     auto record_parser =
-        '<' >> ~as<record>(named_field % ',') >> trailing_comma >> '>'
+        '{' >> ~as<record>(named_field % ',') >> trailing_comma >> '}'
         // Creating a record with repeated field names technically violates
         // the consistency of the underlying stable_map. We live with that
         // until record is refactored into a proper type (FIXME).
-      | ('<' >> (unnamed_field % ',') >> trailing_comma >> '>')
+      | ('{' >> (unnamed_field % ',') >> trailing_comma >> '}')
```
Comment (dominiklohmann) [inline libtenzir/include/tenzir/concept/parseable/tenzir/data.hpp:78]: This should not be part of this PR; I think we need to discuss separately whether we want this now or wait for v5.0 to bundle this with the other planned language changes.

**Comment by jachris** [comment]
FWIW, I can see us solving this problem with expressions instead, solving this without a dedicated operator or by finding a more general solution to the underlying problem. But feel free to continue with this in the meantime.

**Comment by dominiklohmann** [comment]
Closing; we'll handle this as part of the language revamp.
